packages/amplify-provider-awscloudformation/lib/push-resources.js

```diff
@@ -474,6 +484,46 @@ function updateIdPRolesInNestedStack(context, nestedStack, authResourceName) {
 
   Object.assign(nestedStack.Resources, idpUpdateRoleCfn);
 }
+ function packageLayer(context,resource){
+  const resourcePath = path.join(context.amplify.pathManager.getBackendDirPath(), resource.category, resource.resourceName);
+  //const resourcePathSrc = path.join(resourcePath,'src');
+  const layerObj = context.amplify.readJsonFile(path.join(resourcePath,'layer-parameters.json'));
+  const layerVersionArn = layerObj.parameters.layerVersionsArray[0];
+  const zipFilename = 'latest-build.zip';
+
+  // package the function
+  const distDir = path.join(resourcePath, 'dist');
+  if (!fs.existsSync(distDir)) {
+    fs.mkdirSync(distDir);
+  }
+  const destination = path.join(distDir,zipFilename );
+  // verify the folder structure
+
+  // create write stream
+  let zip = ziparchiver.create('zip');
+  let output = fs.createWriteStream(destination);
```

Comment:
It doesn't look like these variables get modified, `const` might be more appropriate

Reply:
Nice catch!! , fixed